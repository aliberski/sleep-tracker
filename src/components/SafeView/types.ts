import { ReactNode } from 'react';
import { ViewStyle } from 'react-native';

export interface IProps {
  children: ReactNode;
}

export interface IStyle {
  container: ViewStyle;
}
