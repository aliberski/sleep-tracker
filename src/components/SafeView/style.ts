import { StyleSheet } from 'react-native';
import { IStyle } from './types';

export default StyleSheet.create<IStyle>({
  container: {
    flex: 1,
    backgroundColor: 'white',
  },
});
