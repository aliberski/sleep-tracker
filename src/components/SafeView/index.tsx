import React from 'react';
import { SafeAreaView } from 'react-navigation';
import { IProps } from './types';

import style from './style';

const SafeView = ({ children }: IProps) => (
  <SafeAreaView style={style.container}>{children}</SafeAreaView>
);

export default SafeView;
