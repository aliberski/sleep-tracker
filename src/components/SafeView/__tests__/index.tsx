import { Text } from 'react-native';
import React from 'react';

import SafeView from '../index';
import renderer from 'react-test-renderer';

describe('SafeView', () => {
  const safeView = renderer.create(
    <SafeView>
      <Text>123</Text>
    </SafeView>,
  );

  it('renders correctly', () => {
    const tree = safeView.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
