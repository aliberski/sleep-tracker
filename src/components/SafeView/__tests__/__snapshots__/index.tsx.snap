// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SafeView renders correctly 1`] = `
<View
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingBottom": 0,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 20,
    }
  }
>
  <Text>
    123
  </Text>
</View>
`;
